cpp/include/cuml/random_projection/rproj_c.h

```diff
@@ -54,15 +57,26 @@ struct rand_mat {
   ~rand_mat() { this->reset(); }
 
   // For dense matrices
-  math_t *dense_data;
+  device_buffer<math_t> *dense_data;
 
   // For sparse CSC matrices
-  int *indices;
-  int *indptr;
-  math_t *sparse_data;
+  device_buffer<int> *indices;
+  device_buffer<int> *indptr;
+  device_buffer<math_t> *sparse_data;
   size_t sparse_data_size;
 
-  void reset();
```

Comment:
This would go away altogether by using the CSR class, and leave less likelihood of leaks

Reply:
This method is also used by the `fit` method of the model. It allows to reset the matrix to `unset` type instead of `dense` or `sparse`. And it also releases unneeded memory, when fitting dense then sparse or conversely. (previously, it also prevented possible leaks caused by overwriting the pointers to the GPU memory)